eval.go

```diff
@@ -431,8 +431,7 @@ nextFileMatch:
 		}
 	})
 
-	// I am slightly worried about negative interactions with TotalMaxMatchCount
-	// so feature flagging this behaviour behind UseDocumentRanks.
+	// If document ranking is enabled, then we can rank and truncate the files to save memory.
```

Comment:
Since we already sort and limit here, I was wondering if we could make `collectSender.Send` more efficient. If it knows that the new files are sorted by score, it could efficiently merge the two sorted lists instead of doing a full sort through `zoekt.SortFiles`.

This comment looked a little concerning though, and I was wondering if I was missing some context here :)

Reply:
I racked my brain a bit, and I think I remember the concern. It was that if your TotalMaxMatchCount is signficantly higher than your MaxDocDisplayCount and you have a lot of shards, we would end up searching all shards since we we wouldn't ever reach TotalMaxMatchCount. Previously we would stop earlier.

However, I actually read the code now and this concern was incorrect. We use stats.MatchCount not the size of Files. And MatchCount is incremented before this step of truncating/sorting. So this is safe :)